lib/topologies/shared.js

```diff
@@ -56,6 +56,20 @@ function createClientInfo(options) {
   return clientInfo;
 }
 
+function createCompressionInfo(options) {
+  if (options.compression && options.compression.compressors) {
```

Comment:
nit: I'm an advocate of bailing early. IMHO it helps readability, and reduces an indentation level. e.g.
```
if (!options.compression || !options.compression.compressors) {
  return [];
}

... the rest ...
```